### PR TITLE
Adding support for auto create and auto delete test container

### PR DESCRIPTION
**Pull request by simplynaveen20**
This PR will create the test db and test container mentioned in the run parameters for all the workload if they doesn't exist.
Also It will delete the container at the end of the test.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java [hunk 1]
```diff
@@ -32,6 +32,7 @@
 
 class Configuration {
 
+    final static String PARTITION_KEY = "/pk";
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java]: this is the default partition-key if the collection does not exist. But if the collection exist the partition-key path could be different and that's ok. right?

So could you please change the naming to `DEFAULT_PARTITION_KEY_PATH`

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncBenchmark.java [hunk 2]
```diff
@@ -130,6 +134,8 @@ protected void init() {
     }
 
     void shutdown() {
+        cosmosAsyncContainer.delete().block();
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncBenchmark.java]: In our testing subscriptions we can't create a collection with 1M throughput upfront. The workaround we have is to create a collection with 100K throughput and then scale up. Now with this auto deleting collection it means we have to do this manual process everytime.

we don't want to always delete the collection. We should have a config option called `autoCollectionCreateEnabled` when present we should attempt to 
1. create collection if not exist 
2. delete the collection

when `autoCollectionCreateEnabled=false` we shouldn't attempt to create or delete the collection, and if the collection doesn't exist we should fail.

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java [hunk 3]
```diff
@@ -227,33 +212,29 @@ private void createClients() {
...
-                            createDocumentObservables.add(obs);
-                        }
-                        docsToRead = Flux.merge(Flux.fromIterable(createDocumentObservables), 100).collectList().block();
-
-                        logger.info("Client have been initialized with data created for host {}", hostAndKey[0]);
-                    } else {
-                        logger.info("Client have been initialized with host {}", hostAndKey[0]);
+                    CosmosAsyncDatabase cosmosAsyncDatabase = asyncClient.createDatabaseIfNotExists(this.configuration.getDatabaseId()).block().getDatabase();
+                    CosmosAsyncContainer cosmosAsyncContainer =
+                        cosmosAsyncDatabase.createContainerIfNotExists(configuration.getCollectionId(), Configuration.PARTITION_KEY, configuration.getThroughput()).block().getContainer();
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java]: if the application deletes it should create too.

when a user runs an application, the user expects to either 1.  application handles the collection creation/deletion 2. the user provides the collection and the app doesn't create nor deletes.

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java (hunk 1 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/Configuration.java]: done

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java (hunk 3 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsynReadWithMultipleClients.java]: Please check current design , it is more seem less in respect to customer expectation , code will only create resource if it is not present , and only delete newly created resource and not the existing

File: sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncBenchmark.java (hunk 2 above)
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/AsyncBenchmark.java]: Please check current design , it is more seem less in respect to customer expectation , code will only create resource if it is not present , and only delete newly created resource and not the existing

**Pull request merged**